File: restful-server/src/main/java/via/sep/restful_server/model/UserProfile.java
package via.sep.restful_server.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "userprofile", schema = "accounts")
@Data
public class UserProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "profile_id")
    private Long profileId;

    @OneToOne
    @JoinColumn(name = "account_id")
    private Login login;

    @Column(name = "full_name", nullable = false, length = 100)
    private String fullName;

    @Column(unique = true, nullable = false, length = 100)
    private String email;

    @Column(name = "phone_number", length = 15)
    private String phoneNumber;

    @Column(length = 255)
    private String address;
}

File: Blazor/RealEstateBlazor/Data/Models/UserProfile.cs
namespace RealEstateBlazor.Data.Models;

public class UserProfile
{
    public long ProfileId { get; set; }
    
    public Login Login { get; set; }
    
    public long AccountId { get; set; }
    
    public string FullName { get; set; }
    
    public string Email { get; set; }
    
    public string PhoneNumber { get; set; }
    
    public string Address { get; set; }
}
